src/librustc/ty/maps/plumbing.rs

```diff
@@ -643,10 +655,23 @@ macro_rules! define_maps {
                     $($name: Lock::new(QueryMap::new())),*
                 }
             }
+
+            pub fn collect_active_jobs(&self) -> Vec<Lrc<QueryJob<$tcx>>> {
+                let mut jobs = Vec::new();
+
+                $(for v in self.$name.lock().active.values() {
```

Comment:
It might be helpful to use `try_lock` here, so that we don't deadlock in the deadlock handler if anybody ever introduces code that accidentally holds a lock to the query tables.

Reply:
Done.